pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -190,6 +193,45 @@ const (
 	KernelTunableModify KernelTunableBehavior = "modify"
 )
 
+// Init creates the top level qos cgroup containers
+// We currently create containers for the Burstable and Best Effort containers
+// All guaranteed pods are nested under the RootContainer by default
+// Init is called only once during kubelet bootstrapping.
+func InitQOS(rootContainer string) (QOSContainersInfo, error) {
+	cm := NewCgroupManagerImpl()
+	// The rootContainer under which all qos containers are brought up is configurable
+	// and can be specified through the --cgroup-root flag.
+	// We default to the system root / in case no root is specified
+	rootContainerName := defaultContainerRoot
+	if rootContainer != "" {
+		rootContainerName = rootContainer
+	}
+	// Top level for Qos containers are created only for Burstable
+	// and Best Effort classes
+	qosClasses := [2]qos.Tier{qos.Burstable, qos.BestEffort}
+
+	// Create containers for both qos classes
+	for _, qosClass := range qosClasses {
+		//get the containers absolute name
+		absoluteContainerName := rootContainerName + string(qosClass)
```

Comment:
typically, i would expect the qosClass to have also have a toLower so Guaranteed is just guaranteed